### PR TITLE
Update testing.yml

### PR DESCRIPTION
**Pull request by DSilva27**
Caching the dependencies is breaking tests as the code in the repo does not get updated. I also removed clutter in the step that runs the tests

### CODE REVIEW

**Comment by geoffwoollard** [comment]
There are two caches. One for dependencies (caches the environment), and one for data (caches the pulled data).

The both have keys, and get remade when the key is new. The key is hashed from, e.g. the contents of some files. 

So for instance when the pyproject.toml changes, the dependency key changes, and there is a new install.

Or when the downloading script from osf changes, the files are downloaded again.

I think it's worth trying to get this working, otherwise the tests can take quite a bit longer.

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -55,8 +55,5 @@ jobs:
           
       - name: Test with pytest
         run: |
-          pytest tests/test_preprocessing.py
-          pytest tests/test_svd.py
-          pytest tests/test_map_to_map.py
-          pytest tests/test_distribution_to_distribution.py
+          pytest tests/
```
Comment (geoffwoollard) [inline .github/workflows/testing.yml]: also, some of these tests have to happen in a certain order, so they should be explicitly stated

Comment (DSilva27) [reply .github/workflows/testing.yml]: yeah, I reverted that change

**Comment by geoffwoollard** [comment]
here is where the key hash files are set
https://github.com/flatironinstitute/Cryo-EM-Heterogeneity-Challenge-1/blob/main/.github/workflows/testing.yml#L33
https://github.com/flatironinstitute/Cryo-EM-Heterogeneity-Challenge-1/blob/main/.github/workflows/testing.yml#L42

**Comment by DSilva27** [comment]
I just know that it has happened twice that I had to remove all the cache for my tests to pass. The pyproject is not really a good way to define these keys, as the code can change without the code changing. Also, the cache is taking more space than what we are supposed to use (17GB when the max is 10GB), you can see this in Actions -> caches.

**Comment by DSilva27** [comment]
Found something that is supposed to do the dependency caching for you https://github.com/actions/setup-python#caching-packages-dependencies

**Comment by geoffwoollard** [comment]
I think caching the external pip dependencies, with the .toml file as a key hash makes sense. There is no other place that new external dependencies enter.

The tests can uninstall and install the actual code of the project (fast) with 
pip uninstall cryo-em-heterogeneity-challenge-1

**Comment by geoffwoollard** [comment]
> Found something that is supposed to do the dependency caching for you https://github.com/actions/setup-python#caching-packages-dependencies

I'm fine introducing a requirements.txt file for pip.

I think it does not make any sense to have the source code mixed in with the cache - that's not what is desired.

**Comment by DSilva27** [comment]
It does not make sense, because it includes the repo itself as a dependency. It is using old versions of the code instead of the version you are pushing. That is why my test was failing, it was using the version from "main". This has two very dangerous consequences:

1. You think your tests are passing, but they are not really passing because the version you pushed is not being used.
2. You think your tests are not passing, but in reality they don't pass because the code is old. I found this mistakes because I changed the config files (added new keys), so when it validated the config file, it was using an old validator.

**Comment by geoffwoollard** [comment]
> * You think your tests are passing, but they are not really passing because the version you pushed is not being used.

I agree that we don't want to cache the source.

But are you sure that when you install, it uses main? It would use whatever was in the cache, if it was using a cache. But if it reinstalls it, it should be from whatever branch the PR is from. Right???

**Comment by DSilva27** [comment]
It uses whatever was cached. If you wan with "branch1234" and no cache exists, then the cache is created with whatever is in that branch. In this case the cache was from main (or dev). Right now it must be from my branch iss14 (not in this branch because I removed the cache).

**Comment by geoffwoollard** [comment]
Ok let's turn off the dependency caching immediately (this PR), and look into caching just the pip requests. It will add about 2-3 min on each check. The testing can be done locally for faster turn around.

**Review by geoffwoollard — APPROVED** [review]

**Comment by geoffwoollard** [comment]
@DSilva27 I approved. You can merge in if it's all good to go. This should be merged into main immediately after it's merged into dev.

**Comment by DSilva27** [comment]
I will merge, but I will not delete the branch just in case.

**Pull request merged**